system_tests/estimation_test.go

```diff
@@ -127,6 +127,46 @@ func TestEstimate(t *testing.T) {
 	}
 }
 
+func TestDifficultyForLatestArbOS(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	l2info, node, client := CreateTestL2(t, ctx)
+	defer node.StopAndWait()
+
+	auth := l2info.GetDefaultTransactOpts("Owner", ctx)
+
+	// deploy a test contract
+	_, _, simple, err := mocksgen.DeploySimple(&auth, client)
+	Require(t, err, "could not deploy contract")
+
+	difficulty, err := simple.GetBlockDifficulty(&bind.CallOpts{})
+	Require(t, err)
+	if arbmath.BigEquals(difficulty, common.Big1) {
+		Fail(t, "Expected difficulty to be 1 by got:", difficulty)
+	}
+}
+
+func TestDifficultyForArbOSTen(t *testing.T) {
```

Comment:
let's do that with our plan for block_validator test.
Would be interesting to see what we get for difficulty right before and right after the upgrade (in the same block that we upgrade), and then in a separate block.

Reply:
done.